### ISSUE TITLE
Declare real variables as `real(kind=dp)`

### ISSUE THREAD

**Issue by Beliavsky** [ChristianSteger/grid_interpolation#1]
Your codes use default real variables. I think it would be better to declare the variables
as double precision with code such as

```
integer, parameter :: dp = kind(1.0d0)
real(kind=dp) :: x, y
```
especially since Python floating point variables are double precision.

**Comment by ChristianSteger** [comment]
Good point - these routines will be use to couple two models in earth/climate science that run on different grids. However, we first have to check if these models run in single/double precision before I consider the above suggestion - I thus will leave this open for the moment...

**Comment by Beliavsky** [comment]
Even if the the models use single precision, you can write

```
integer, parameter :: wp = kind(1.0)
real(kind=wp) :: x, y
```
It's best if the code is written so that all variables can be changed to single or double precision by changing one line, in this case the line that sets `wp`.
